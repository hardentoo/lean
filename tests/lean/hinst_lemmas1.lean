#exit
axiom foo1 : ∀ (a b c : nat), b > a → b < c  → a < c
axiom foo2 : ∀ (a b c : nat), b > a → b < c → a < c
axiom foo3 : ∀ (a b c : nat), b > a → b < c + c → a < c + c


run_cmd
do
  hs ← return $ hinst_lemmas.mk,
  h₁ ← hinst_lemma.mk_from_decl `foo1,
  h₂ ← hinst_lemma.mk_from_decl_core tactic.transparency.none `foo2 ff,
  h₃ ← hinst_lemma.mk_from_decl `foo3,
  hs ← return $ ((hs^.add h₁)^.add h₂)^.add h₃,
  hs^.pp >>= tactic.trace
