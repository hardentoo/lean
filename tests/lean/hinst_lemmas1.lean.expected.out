hinst_lemmas1.lean:1:0: warning: using 'exit' to interrupt Lean
